components/Rate/__test__/rate.spec.ts

```diff
@@ -0,0 +1,299 @@
+import { afterEach, expect, test, describe, beforeEach } from 'vitest';
+import KRate from '../src';
+import { tick } from 'svelte';
+
+let host: HTMLElement;
+
+const initHost = () => {
+	host = document.createElement('div');
+	host.setAttribute('id', 'host');
+	document.body.appendChild(host);
+};
+beforeEach(() => {
+	initHost();
+});
+afterEach(() => {
+	host.remove();
+});
+
+describe('Test: KRate', () => {
+	test('props: cls', async () => {
+		const instance = new KRate({
+			target: host,
+			props: {
+				cls: 'k-rate--test'
+			}
+		});
+		expect(instance).toBeTruthy();
+		expect((host as HTMLElement)!.innerHTML.includes('k-rate--test')).toBeTruthy();
+		expect(host.innerHTML).matchSnapshot();
+	});
+
+	test('props: max', async () => {
+		const max = 10;
+		const instance = new KRate({
+			target: host,
+			props: {
+				max
+			}
+		});
+		expect(instance).toBeTruthy();
+		expect(host.querySelectorAll('.k-rate--item').length).toBe(max);
+		expect(host.innerHTML).matchSnapshot();
+	});
+
+	test('props: value', async () => {
+		const value = 3;
+		const instance = new KRate({
+			target: host,
+			props: {
+				value
+			}
+		});
+		expect(instance).toBeTruthy();
+		expect(host.querySelectorAll('.i-carbon-star-filled').length).toBe(value);
+		expect(host.innerHTML).matchSnapshot();
+	});
+
+	test('props: showScore', async () => {
+		const value = 3;
+		const showScore = true;
+		const instance = new KRate({
+			target: host,
+			props: {
+				value,
+				showScore
+			}
+		});
+		expect(instance).toBeTruthy();
+		expect(host.querySelector('.k-rate--text')?.textContent).toBe(`${value}`);
+		expect(host.innerHTML).matchSnapshot();
+	});
+
+	test('props: scoreTemplate', async () => {
+		const value = 3;
+		const showScore = true;
+		const scoreTemplate = 'this is {value}';
+		const instance = new KRate({
+			target: host,
+			props: {
+				value,
+				showScore,
+				scoreTemplate
+			}
+		});
+		expect(instance).toBeTruthy();
+		expect(host.querySelector('.k-rate--text')?.textContent).toBe(`this is ${value}`);
```

Comment:
We are not sure if there is a problem with `const scoreTemplate = 'this is {value}';` and `toBe(`this is ${value}`);` , please provide some confirmation information when you are done.

Reply:
> We are not sure if there is a problem with `const scoreTemplate = 'this is {value}';` and `toBe(`this is ${value}`);` , please provide some confirmation information when you are done.

I may need to reconstruct the icon component. Based on the current component implementation, I have no idea of ​​​​implementing half-selection for the time being.